python/dgl/distributed/graph_services.py

```diff
@@ -504,12 +538,29 @@ def sample_etype_neighbors(g, nodes, etype_field, fanout, edge_dir='in',
             homo_nids.append(gpb.map_to_homo_nid(typed_nodes, ntype))
         nodes = F.cat(homo_nids, 0)
     def issue_remote_req(node_ids):
-        return SamplingRequestEtype(node_ids, etype_field, fanout, edge_dir=edge_dir,
-                                    prob=prob, replace=replace, etype_sorted=etype_sorted)
+        if prob is not None:
+            # See NOTE 1
+            # Is there a better way to get the KVStore key given the name and
+            # node/edge type?
+            _prob = [g.edges[etype].data[prob].kvstore_key for etype in g.etypes]
+        else:
+            _prob = None
+        return SamplingRequestEtype(
+                node_ids, etype_field, eid_field, fanout, edge_dir=edge_dir,
+                prob=_prob, replace=replace, etype_sorted=etype_sorted)
     def local_access(local_g, partition_book, local_nids):
-        return _sample_etype_neighbors(local_g, partition_book, local_nids,
-                                       etype_field, fanout, edge_dir, prob, replace,
-                                       etype_sorted=etype_sorted)
+        # See NOTE 1
+        if prob is None:
+            return _sample_etype_neighbors(
+                    local_g, partition_book, local_nids, etype_field,
+                    eid_field, fanout, edge_dir, None, replace,
+                    etype_sorted=etype_sorted)
+        else:
+            _prob = [g.edges[etype].data[prob].local_partition for etype in g.etypes]
+            return _sample_etype_neighbors(
+                    local_g, partition_book, local_nids, etype_field,
+                    eid_field, fanout, edge_dir, _prob, replace,
+                    etype_sorted=etype_sorted)
```

Comment:
L554-L557 and L560-L563 are very similar. Can be merged with an `if` branch to get the right `_prob` array.